packages/common/selectors/keyStore.js

```diff
@@ -19,11 +27,29 @@ export const personalKeyPairSelector = createSelector(
 export const teamKeyPairSelector = createSelector(
   keyStoreDataSelector,
   teamIdPropSelector,
-  (data, teamName) =>
-    Object.values(data[KEY_TYPE.TEAM]).find(({ name }) => name === generateSystemItemName(teamId)) || {},
+  (data, teamId) => findTeamItemByName(data, teamId),
 );
 
 export const anonymousKeyPairSelector = createSelector(
   keyStoreDataSelector,
   (data, teamName) => Object.values(data[KEY_TYPE.ANONYMOUS]) || {},
 );
+
+export const actualKeyPairSelector = createSelector(
+  keyStoreDataSelector,
+  currentTeamIdSelector,
+  personalKeyPairSelector,
+  isUserAnonymousSelector,
+  userIdSelector,
+  (data, currentTeamId, personalKeyPair, isAnonymous, userId) => {
+    if (isAnonymous) {
+      return Object.values(data[KEY_TYPE.ANONYMOUS]).find(({ id }) => userId) || {};
+    }
+
+    if (currentTeamId !== TEAM_TYPE.PERSONAL) {
+      return findTeamItemByName(data, currentTeamId);
+    }
+
+    return personalKeyPair;
```

Comment:
// nitpick
because of sensitive data I'll better add condition
`if (currentTeamId === TEAM_TYPE.PERSONAL) return personalKeyPair `.
And default - null.
To avoid receive sensitive data.

And think that it would be better to use switch case construction, when more than 1 condition.

Reply:
Done